pycloudlib/oci/cloud.py

```diff
@@ -343,3 +344,27 @@ def snapshot(self, instance, clean=True, name=None):
         self.created_images.append(image_data.id)
 
         return image_data.id
+
+    @staticmethod
+    def _validate_tag(tag: str):
+        """
+        Ensure that this tag is a valid name for cloud resources.
+
+        Rules:
+        - Must be between 1 and 255 characters long, after stripping whitespace
+
+        :param tag: tag to validate
+
+        :raises InvalidTagNameError: if the tag is invalid
+        """
+        tag = tag.strip()
```

Comment:
Do we actually want to strip whitespace here? If it's not also stripped in the parent, then isn't it possible we'd pass here but fail on actual creation? If it is stripped in the parent, can we move the stripping earlier so that it happens before being passed here?

Reply:
Oh shoot yeah good catch.